xarray/plot/facetgrid.py

```diff
@@ -343,6 +343,7 @@ def add_legend(self, **kwargs):
             title=self._hue_label,
             loc="center right", **kwargs)
 
+        self.figlegend = figlegend
```

Comment:
I needed this to write a unit test that checks the text in the legend. I think it's good as it gives the user an easy way to directly access the generated legend. But Since this is technically an API change,  I thought maybe I should stress that when you guys review the code.

Reply:
Can you use something like https://stackoverflow.com/questions/24140196/how-to-get-all-legends-from-a-plot instead? Or maybe save it to a private attribute instead?

Reply:
I thought about that. I don't think it's difficult, but I also don't see why not expose the legend to the end users so they won't have to go to stackexchange to understand how to get the legend...

Reply:
That's fair enough... but in that case, can we be sure to set it in `__init__`, too? Something like `self.figlegend = None`